tempesta_fw/http_msg.c

```diff
@@ -573,12 +577,62 @@ tfw_http_msg_hdr_xfrm(TfwHttpMsg *hm, char *name, size_t n_len,
 }
 
 /**
+ * Remove hop-by-hop headers in the message
+ *
+ * Connection header should not be removed, tfw_http_set_hdr_connection()
+ * optimize removal of the header.
+ */
+int
+tfw_http_msg_del_hbh_hdrs(TfwHttpMsg *hm)
+{
+	TfwHttpHdrTbl *ht = hm->h_tbl;
+	unsigned int hid = ht->off;
+	int r = 0;
+
+	do {
+		hid--;
+
+		if (hid == TFW_HTTP_HDR_CONNECTION)
+			continue;
+		if (ht->tbl[hid].flags & TFW_STR_HBH_HDR)
+			if ((r = __hdr_del(hm, hid)))
+				return r;
+	} while (hid != 0);
+
+	return 0;
+}
+
+/**
+ * Lookup for the header @hdr in already collected headers table @ht,
+ * and mark it as hop-by-hop. The lookup is performed untill ':', so header
+ * name only is enough in @hdr.
+ */
+void
+tfw_http_msg_mark_hbh_hdr(TfwHttpMsg *hm, TfwStr *hdr)
```

Comment:
I believe that this function (under a better name) belongs in `http_parser.c`.
If I understand correctly, the reason for placing it here was that `__hdr_lookup()` was located here, and it's a local (static) function to this sub-module. In my opinion though it should be the other way around: `__hdr_lookup()` should be made public (and given a proper name for that) and used from `http_parser.c`